components/map.jsx

```diff
@@ -21,76 +23,143 @@ function geoJSONit(data) {
+// it can't have any dynamic functionality.
+//
+// Furthermore, because we aren't a "live" React element, events
+// dispatched from the map will need to be processed when they bubble
+// up to the parent map component.
+var MarkerPopup = React.createClass({
+  getWebsiteDomain: function() {
+    return urlParse(this.props.website).hostname;
+  },
+  render: function() {
+    var actions = null;
+
+    if (this.props.isOwned) {
+      actions = (
+        <div>
+          <button className="btn btn-default btn-xs"
+           data-club-action="edit" data-club-url={this.props.url}>
+            <span className="glyphicon glyphicon-pencil"></span> Edit
+          </button>
+          &nbsp;
+          <button className="btn btn-default btn-xs"
+           data-club-action="delete" data-club-url={this.props.url}
+           data-club-name={this.props.title}>
+            <span className="glyphicon glyphicon-trash"></span> Remove
+          </button>
+        </div>
+      );
+    }
+
+    return (
+      <div>
+        <b>{this.props.title}<br/></b>
+        <i>{this.props.location}</i>
+        <br/>
+        <br/>
+        <p>{this.props.description}</p>
+        <p><a href={this.props.website} target="_blank">
+          {this.getWebsiteDomain()}
+        </a></p>
+        {actions}
+      </div>
+    );
+  }
+});
+
 var Map = React.createClass({
-  mixins: [TeachAPIClientMixin],
   propTypes: {
     className: React.PropTypes.string.isRequired
   },
   componentDidMount: function() {
-    var that = this;
     require('mapbox.js'); // this will automatically attach to Window object.
     require('leaflet.markercluster');
     L.mapbox.accessToken = accessToken;
-    this.getTeachAPI().getAllClubsData(function(err, data) {
-      if (!that.isMounted()) {
-        // We were unmounted before the API request completed,
-        // so do nothing.
-        return;
-      }
-      if(err) {
-        console.log(err)
-        return;
+    this.map = L.mapbox.map(this.getDOMNode())
+      .setView([40.73, -50.011], 5)
+      .addLayer(L.mapbox.tileLayer(mapboxId));
+    this.markers = new L.MarkerClusterGroup({
+      iconCreateFunction: function(cluster) {
+        return L.mapbox.marker.icon({
+          'marker-symbol': cluster.getChildCount(),
+          'marker-color': '#422'
+        });
       }
-      var geoJSON = geoJSONit(data);
-      that.map = L.mapbox.map(that.getDOMNode())
-        .setView([40.73, -50.011], 5)
-        .addLayer(L.mapbox.tileLayer(mapboxId));
-      var markers = new L.MarkerClusterGroup({
-        iconCreateFunction: function(cluster) {
-          return L.mapbox.marker.icon({
-            'marker-symbol': cluster.getChildCount(),
-            'marker-color': '#422'
-          });
-        }
-      });
+    });
 
-      // setView of the map based on markers on the map
-      that.map.fitBounds(L.mapbox.featureLayer()
-        .setGeoJSON(geoJSON)
-        .getBounds());
+    this.map.fitBounds([[65, 0], [-65, 0]]);
```

Comment:
Oh, sorry, I did that because we're no longer actually rendering the map with all the markers known at instantiation time...  Originally the map will be given a `clubs` prop that is an empty array, because we're still asking the Teach API for a list of markers, but once we have some entries, the component will update. That means we have to have some kind of initial viewport, and since there will be clubs around the world, I figured it'd be best to just show the whole map.

Another alternative, I think, is to just set the initial viewport via the mapbox UI, on the `alicoding.ldmhe4f3` map... I *think* that whatever its initial viewport is, will be the initial viewport that the map widget uses if we don't override it like we're doing here.

Um, I think we're also overriding it on line 81, actually--so we should probably get rid of line 92 regardless!